views/pages/ViewChartPage.jsx

```diff
@@ -47,11 +60,16 @@ const ViewChartPage = () => {
   }
 
   useEffect(() => {
-    fetchGraph(chart_id).then((graph) => {
-      setGraph(graph)
-      reset(graph.filters)
+    const parsedQuery = qs.parse(search.replace(/^\?/, ''))
+
+    fetchGraph(chart_id, parsedQuery.filters).then((newGraph) => {
+      setGraph(newGraph)
+      reset({
+        ...newGraph.filters,
+        sites: StudiesModel.dropdownSelectOptions(newGraph.filters.sites),
```

Comment:
This logic is being used twice, I wonder if we can assign it to a variable..?

Reply:
what would that look like? I see that we're using it twice but it seems like we need to (unless we don't need to call `reset`)? Do you know why we need to reset the form?

Reply:
I think with what we did with the service we can do it all from the useEffect

Reply:
reset is used when we need to hydrate form values from an async call

Reply:
that service is on the server side, so we know the data is accurate we just need to convert it to dropdown options.